### PR TITLE
Set up basic CI

### PR DESCRIPTION
**Pull request by janpaul123**
Basic CI using Github Actions, which is free for public repos.

This still requires manually pushing a new docker image whenever Dockerfile-ci changes. I tried to automate that within the CI pipeline, but that seems a little tricky. We'll save that for another day.

Best viewed commit-by-commit.

### CODE REVIEW

File: zaplib/examples/example_single_button/src/single_button.rs [hunk 1]
```diff
@@ -46,11 +46,11 @@ Equivalent React functional-style:
 
 function SingleButton() {
     const [clicks, setClicks] = useState(false);
-    
+
     const onButtonClick = useCallback(() => {
         setClicked(clicks + 1)
     }, [clicks, setClicks]);
-
```
Comment (stevekrouse) [inline zaplib/examples/example_single_button/src/single_button.rs:53]: Do you have a rust formatter? Can you add instructions for adding somewhere in the readme?

**Review by stevekrouse — APPROVED** [review]
This PR is way over my head. Is there anything in particular you want my extra set of eyes on?

**Review by hhsaez — APPROVED** [review]
This looks good to me and all actions are passing. Nice!

**Comment by janpaul123** [comment]
> Is there anything in particular you want my extra set of eyes on?

Nope your comment was already super helpful!

**Pull request merged**